Проблема при сборке cmake с perl

При сборке с помощью cmake вываливается следующая ошибка и hpt не собирается:
```
[fido@srv-srt2 hpt]$ cmake -H. -Bbuild -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX:PATH=/home/fido/fido_make/husky
-------------------------
System            : Linux 4.18.0-147.3.1.el8_1.x86_64
Compiler          : GNU 8.3.1
Prefix            : /home/fido/fido_make/husky (run "cmake . -DCMAKE_INSTALL_PREFIX:PATH=/other" for other prefix)
libhusky          : /home/fido/fido_make/husky/lib64/libhusky.so
libfidoconfig     : /home/fido/fido_make/husky/lib64/libfidoconfig.so
libsmapi          : /home/fido/fido_make/husky/lib64/libsmapi.so
libareafix        : /home/fido/fido_make/husky/lib64/libareafix.so
hptzip            : hptzip_LIB-NOTFOUND
perl              : TRUE
Build shared      : ON
Build type        : Release
-------------------------
-- Configuring done
-- Generating done
-- Build files have been written to: /home/fido/fidosoft/husky/hpt/build
[fido@srv-srt2 hpt]$ cmake --build build
[ 17%] Built target tpkt
[ 24%] Built target hptlink
[ 26%] Building C object CMakeFiles/hpt.dir/src/perl.c.o
In file included from /home/fido/fidosoft/husky/hpt/src/perl.c:73:
/usr/lib64/perl5/CORE/perl.h:2727:22: ошибка: неизвестное имя типа <<off64_t>>; имелось в виду <<off_t>>?
 #       define Off_t off64_t
                      ^~~~~~~
/usr/lib64/perl5/CORE/perlio.h:248:15: замечание: в расширении макроса <<Off_t>>
 PERL_CALLCONV Off_t PerlIO_tell(PerlIO *);
               ^~~~~
/usr/lib64/perl5/CORE/perl.h:2727:22: ошибка: неизвестное имя типа <<off64_t>>; имелось в виду <<off_t>>?
 #       define Off_t off64_t
                      ^~~~~~~
/usr/lib64/perl5/CORE/perlio.h:251:41: замечание: в расширении макроса <<Off_t>>
 PERL_CALLCONV int PerlIO_seek(PerlIO *, Off_t, int);
                                         ^~~~~
/usr/lib64/perl5/CORE/perl.h:2727:22: ошибка: неизвестное имя типа <<off64_t>>; имелось в виду <<off_t>>?
 #       define Off_t off64_t
                      ^~~~~~~
/usr/lib64/perl5/CORE/proto.h:791:47: замечание: в расширении макроса <<Off_t>>
 PERL_CALLCONV bool Perl_do_seek(pTHX_ GV* gv, Off_t pos, int whence);
                                               ^~~~~
/usr/lib64/perl5/CORE/perl.h:2727:22: ошибка: неизвестное имя типа <<off64_t>>; имелось в виду <<off_t>>?
 #       define Off_t off64_t
                      ^~~~~~~
/usr/lib64/perl5/CORE/proto.h:798:15: замечание: в расширении макроса <<Off_t>>
 PERL_CALLCONV Off_t Perl_do_sysseek(pTHX_ GV* gv, Off_t pos, int whence);
               ^~~~~
/usr/lib64/perl5/CORE/perl.h:2727:22: ошибка: неизвестное имя типа <<off64_t>>; имелось в виду <<off_t>>?
 #       define Off_t off64_t
                      ^~~~~~~
/usr/lib64/perl5/CORE/proto.h:798:51: замечание: в расширении макроса <<Off_t>>
 PERL_CALLCONV Off_t Perl_do_sysseek(pTHX_ GV* gv, Off_t pos, int whence);
                                                   ^~~~~
/usr/lib64/perl5/CORE/perl.h:2727:22: ошибка: неизвестное имя типа <<off64_t>>; имелось в виду <<off_t>>?
 #       define Off_t off64_t
                      ^~~~~~~
/usr/lib64/perl5/CORE/proto.h:801:15: замечание: в расширении макроса <<Off_t>>
 PERL_CALLCONV Off_t Perl_do_tell(pTHX_ GV* gv)
               ^~~~~
/usr/lib64/perl5/CORE/perl.h:2727:22: ошибка: неизвестное имя типа <<off64_t>>; имелось в виду <<off_t>>?
 #       define Off_t off64_t
                      ^~~~~~~
/usr/lib64/perl5/CORE/proto.h:6090:53: замечание: в расширении макроса <<Off_t>>
 PERL_CALLCONV int Perl_PerlIO_seek(pTHX_ PerlIO *f, Off_t offset, int whence);
                                                     ^~~~~
/usr/lib64/perl5/CORE/perl.h:2727:22: ошибка: неизвестное имя типа <<off64_t>>; имелось в виду <<off_t>>?
 #       define Off_t off64_t
                      ^~~~~~~
/usr/lib64/perl5/CORE/proto.h:6103:15: замечание: в расширении макроса <<Off_t>>
 PERL_CALLCONV Off_t Perl_PerlIO_tell(pTHX_ PerlIO *f);
               ^~~~~
gmake[2]: *** [CMakeFiles/hpt.dir/build.make:245: CMakeFiles/hpt.dir/src/perl.c.o] Ошибка 1
gmake[1]: *** [CMakeFiles/Makefile2:142: CMakeFiles/hpt.dir/all] Ошибка 2
gmake: *** [Makefile:152: all] Ошибка 2
```

Система:
```
[fido@srv-srt2 hpt]$ cat /etc/redhat-release
Red Hat Enterprise Linux release 8.1 (Ootpa)
```

с помощью cd hpt && make clean all install всё собирается нормально.

--- a/.travis.yml
+++ b/.travis.yml
@@ -25,7 +25,7 @@ matrix:
     dist: xenial
     env:
     - MATRIX_EVAL="CC=gcc" CF="-DBUILD_SHARED_LIBS=ON" ARCH="x86_64"
-      CFLAGS="-Doff64_t=__off64_t" DEPLOY="false"
+      DEPLOY="false"
 before_install:
 - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get install -y rpm; fi
 - git clone https://github.com/huskyproject/huskylib ~/build/huskyproject/huskylib

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,8 @@ PROJECT(hpt C)
 set(areafix_VERSION 1.9.0)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
+include(GNUInstallDirs)
+
 option(BUILD_SHARED_LIBS "Build shared libs" ON)
 
 if(CMAKE_BUILD_TYPE MATCHES Debug)
@@ -102,6 +104,8 @@ target_link_libraries(hpttree ${fidoconfig_LIB} ${smapi_LIB} ${husky_LIB})
 target_link_libraries(hpt ${areafix_LIB} ${fidoconfig_LIB} ${smapi_LIB}  ${husky_LIB})
 
 if (${PERLLIBS_FOUND})
+  EXECUTE_PROCESS(COMMAND ${PERL_EXECUTABLE} -e "use Config; print \$Config{ccflags}" OUTPUT_VARIABLE PERL_CCFLAGS)
+  ADD_DEFINITIONS(${PERL_CCFLAGS})
   if (NOT MSVC)
     target_link_libraries(hpt -lpthread)
   endif()
@@ -124,7 +128,7 @@ endif(hptzip_LIB)
 
 INSTALL(TARGETS pktinfo tpkt txt2pkt hptlink hpttree hpt
   RUNTIME DESTINATION bin
-  LIBRARY DESTINATION lib
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
 
 if (MSVC)

